docs/dataset/formatting-your-datasets.md

```diff
@@ -0,0 +1,159 @@
+| [`ImageAsset`](../reference/asset.md#kolena.asset.ImageAsset)           | Useful if your data is modeled as multiple related images.     |
+| [`BinaryAsset`](../reference/asset.md#kolena.asset.BinaryAsset)         | Useful if you want to attach any segmentation or bitmap masks. |
+| [`AudioAsset`](../reference/asset.md#kolena.asset.AudioAsset)           | Useful if you want to attach an audio file.                    |
+| [`VideoAsset`](../reference/asset.md#kolena.asset.VideoAsset)           | Useful if you want to attach a video file.                     |
+| [`PointCloudAsset`](../reference/asset.md#kolena.asset.PointCloudAsset) | Useful for attaching 3D point cloud data.                      |
+
+### Kolena Annotations
+
+Kolena allows you to visualize overlays on top of datapoints through the use of[`annotation`](../reference/annotation.md).
+These annotations are visible on both the Gallery view for groups of datapoints and for individual datapoints.
+
+| Annotation Type                                                                      | Description |
+|--------------------------------------------------------------------------------------|----------------------------|
+| [`BoundingBox`](../reference/annotation.md#kolena.annotation.BoundingBox)            | Used to overlay bounding boxes (including confidence scores and labels) on top of images. |
+| [`SegmentationMask`](../reference/annotation.md#kolena.annotation.SegmentationMask)  | Used to overaly raster segmentation maps on top of images. |
+
+### Structured Data
+
+When uploading `.csv` files for datasets that contain annotations, assets or nested values in a column use the
+[`dataframe_to_csv()`](../reference/io.md#kolena.io.dataframe_to_csv) function provided by Kolena to save a `.csv` file
+instead of [`pandas.to_csv()`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_csv.html).
+
+In order to add data like a list of `BoundingBox` objects to your dataset via the sdk all you need to do is have
+field with a list of objects in your dataframe.
+
+A snippet like the following:
+
+```python
+from kolena.annotation import BoundingBox
+from kolena.io import dataframe_to_csv
+from collections import defaultdict
+import pandas as pd
+
+df = pd.read_csv(f"s3://kolena-public-examples/coco-2014-val/raw/coco-2014-val.csv", storage_options={"anon": True})
+image_to_boxes = defaultdict(list)
+image_to_metadata = defaultdict(dict)
+
+for record in df.itertuples():
+    coords = (float(record.min_x), float(record.min_y)), (float(record.max_x), float(record.max_y))
+    bounding_box = BoundingBox(*coords)
+    image_to_boxes[record.locator].append(bounding_box)
+    metadata = {
+            "locator": str(record.locator),
+            "height": float(record.height),
+            "width": float(record.width),
+            "date_captured": str(record.date_captured),
+            "brightness": float(record.brightness),
```

Comment:
Why are these casts necessary?

Reply:
These casts were not necessary, removed. Made some adjusments to this code and was able to create a dataset by uploading the produced csv on the WebApp.